pkg/skaffold/schema/profiles_test.go

```diff
@@ -464,6 +464,49 @@ func TestActivatedProfiles(t *testing.T) {
 	}
 }
 
+func TestYamlAlias(t *testing.T) {
+	config := `
+.activation_common: &activation_common
+  env: ABC=common
+build:
+  artifacts:
+  - image: example
+profiles:
+- name: simple1
+  activation:
+  - *activation_common
+  - env: ABC=1
+- name: simple2
+  activation:
+  - *activation_common
+  - env: ABC=2
+`
+
+	testutil.Run(t, "", func(t *testutil.T) {
+		tmpDir := t.NewTempDir().
+			Write("skaffold.yaml", addVersion(config))
+
+		parsed, err := ParseConfig(tmpDir.Path("skaffold.yaml"), false)
+		t.CheckNoError(err)
+		if parsed == nil || err != nil {
```

Comment:
t.CheckNoError(err) should be enough here to test that there's no error

Reply:
It would surely fail the test as expected, but the issue is that the test code keeps running after `CheckNoError`. Comment out the `if parsed == nil || err != nil {` _here_ and the "ignore dotted keys" feature, and you will get failure output from the line `skaffoldConfig := parsed.(*latest.SkaffoldConfig)` which should not even run:

```
Running tool: /usr/local/Cellar/go/1.12.9/libexec/bin/go test -timeout 30s github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema -run ^(TestYamlAlias)$

panic: interface conversion: util.VersionedConfig is nil, not *latest.SkaffoldConfig [recovered]
	panic: interface conversion: util.VersionedConfig is nil, not *latest.SkaffoldConfig

goroutine 23 [running]:
testing.tRunner.func1(0xc0003f6300)
	/usr/local/Cellar/go/1.12.9/libexec/src/testing/testing.go:830 +0x392
panic(0x18f4f00, 0xc0001ce060)
	/usr/local/Cellar/go/1.12.9/libexec/src/runtime/panic.go:522 +0x1b5
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema.TestYamlAlias.func1(0xc00015c120)
	/Users/asommer/dev/extern/skaffold/pkg/skaffold/schema/profiles_test.go:496 +0x5b4
github.com/GoogleContainerTools/skaffold/testutil.Run.func1(0xc0003f6300)
	/Users/asommer/dev/extern/skaffold/testutil/util.go:122 +0x89
testing.tRunner(0xc0003f6300, 0xc0001e00f0)
	/usr/local/Cellar/go/1.12.9/libexec/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.12.9/libexec/src/testing/testing.go:916 +0x35a
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema	0.034s
Error: Tests failed.
```

Since that's confusing for test developers, I suggest introducing a new helper `RequireNoError` which panics immediately (other test frameworks also distinguish "CHECK_*" and "REQUIRE_*" in this way).

Reply:
I agree that sometimes we want that behavior - `testing.T` has `FailNow`, which we could call - I don't think panic is a good idea - that stops the whole testsuite

Reply:
Added such a function (using `t.FailNow()`), and rebased onto master